sidekick_vault/lib/src/commands/vault_command.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sidekick_vault/src/gpg.dart';
+
+class VaultCommand extends Command {
+  VaultCommand() {
+    addSubcommand(_EncryptCommand());
+    addSubcommand(_DecryptCommand());
+  }
+
+  @override
+  String get description => 'Manages sidekick vault';
+
+  @override
+  String get name => 'vault';
+}
+
+class _EncryptCommand extends Command {
+  @override
+  String get description => 'Encrypts a file using gpg';
+
+  @override
+  String get name => 'encrypt';
+
+  @override
+  Future<void> run() async {
+    final file = _validateInput(
+      argResults,
+      example: 'noa vault encrypt file.csv',
+    );
+    final password = ask('Enter password:', hidden: true);
+    file.encrypt(password);
+    print(green('Successfully encrypted $file'));
+  }
+}
+
+class _DecryptCommand extends Command {
+  @override
+  String get description => 'Decrypts a file using gpg';
+
+  @override
+  String get name => 'decrypt';
+
+  @override
+  Future<void> run() async {
+    final file = _validateInput(
+      argResults,
+      example: 'noa vault decrypt file.csv.gpg',
+    );
+    final password = ask('Enter password:', hidden: true);
+    file.decrypt(password);
+    print(green('Successfully decrypted $file'));
+  }
+}
+
+File _validateInput(ArgResults? input, {required String example}) {
```

Comment:
To me `_parseArguments` fits better.
I don't expect a `validate` method to return a value and I associate `input` with stdin, not with command arguments